core_v2/sources/domains.move

```diff
@@ -749,6 +770,33 @@ module aptos_names_v2::domains {
         }
     }
 
+    /// Burns the ANS token v1, mints ANS token v2, and extends the expiration by a year.
+    public entry fun migrate_domain_from_v1(
+        user: &signer,
+        domain_name: String,
+    ) acquires CollectionCapabilityV2, NameRecordV2, RegisterNameEventsV1, ReverseRecord, SetNameAddressEventsV1, SetReverseLookupEventsV1 {
+        let (expiration_time_sec, target_addr) = migrate_helper::burn_token_v1(
+            user,
+            &get_burn_signer(),
+            domain_name,
+            option::none(),
+        );
+        let new_expiration_time_sec = expiration_time_sec + time_helper::years_to_seconds(1);
+        let now = timestamp::now_seconds();
+        assert!(new_expiration_time_sec >= now, error::invalid_state(EMIGRATION_ALREADY_EXPIRED));
```

Comment:
if the original expiration date passed, should we allow them to migrate? I think we should check the original date instead of the extended date?

Reply:
+1 unless we want to give some grace period, otherwise i think if original expiration time > now then error